### ISSUE TITLE
Kpt should accept colons in RBAC resource names

### ISSUE THREAD

**Issue by mgoltzsche** [GoogleContainerTools/kpt#899]
Apparently kpt validates object names and does not accept resource names that contain a colon.
However at least `Role`, `RoleBinding`, `ClusterRole` and `ClusterRoleBindings` can have a name containing a colon - actually very common.  

This can be reproduced by e.g. trying to deploy cert-manager (using kpt 0.32.0):
```
$ mkdir cert-manager && cd cert-manager
$ curl -fsSL https://github.com/jetstack/cert-manager/releases/download/v0.16.0/cert-manager.yaml > cert-manager.yaml
$ kpt live init .
$ kpt live apply .
Fatal error: invalid characters in object name: cert-manager-cainjector:leaderelection
```

kpt's resource name validation should be aligned with k8s`.

**Comment by seans3** [comment]
According to Kubernetes documentation: `Most resource types require a name that can be used as a DNS subdomain name as defined in RFC 1123`. 

https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

This means resource names have the following characteristics:

* contain no more than 253 characters
* contain only lowercase alphanumeric characters, '-' or '.'
* start with an alphanumeric character
* end with an alphanumeric character

According to these documents, colon is not a valid character for a resource name.

**Comment by seans3** [comment]
I'll leave this open for a bit. If there are no responses, I will shortly close this.

**Comment by mgoltzsche** [comment]
@seans3 I also thought so but have a look at your RBAC resources yourself:
```
$ kubectl get -n kube-system role
NAME                                             CREATED AT
cert-manager-cainjector:leaderelection           2020-08-04T00:44:33Z
cert-manager:leaderelection                      2020-08-04T00:44:33Z
extension-apiserver-authentication-reader        2020-08-04T00:44:20Z
kube-proxy                                       2020-08-04T00:44:23Z
kubeadm:kubelet-config-1.18                      2020-08-04T00:44:21Z
kubeadm:nodes-kubeadm-config                     2020-08-04T00:44:21Z
system::leader-locking-kube-controller-manager   2020-08-04T00:44:20Z
system::leader-locking-kube-scheduler            2020-08-04T00:44:20Z
system:controller:bootstrap-signer               2020-08-04T00:44:20Z
system:controller:cloud-provider                 2020-08-04T00:44:20Z
system:controller:token-cleaner                  2020-08-04T00:44:20Z
```

**Comment by seans3** [comment]
You are correct. I was wrong. Indeed colon is allowed clusterrole, role, and the bindings:

https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/rbac/validation/validation.go#L32

I will look into how to allow these in kpt.

**Comment by seans3** [comment]
The problem: 
* We need to store the namespace/name/GVK of each resource as a data in a ConfigMap (as references in the inventory object)
* The ConfigMap data must be valid label (RFC 1123)
* Currently, the separation character is `_`, which is allowed in labels, but not in RFC 1123 resource names
* ClusterRole/Role/Bindings have characters which do not validate as a label, so they can not be currently stored in inventory objects. We can do some special processing for `:` (e.g. `:` == `__`). But the ClusterRole/Role/Bindings also allow numerous other RFC 1123 invalid characters (e.g. `\`, upper-case chars, ...)
